paymentsheet/src/main/java/com/stripe/android/paymentsheet/model/PaymentOption.kt

```diff
@@ -73,166 +55,13 @@ data class PaymentOption internal constructor(
      */
     fun icon(): Drawable {
         return DelegateDrawable(
-            ShapeDrawable(),
             imageLoader,
-            this,
             delegateDrawableScope,
             delegateDrawableDispatcher
         )
     }
 }
 
-private val errorImageLoader: suspend (PaymentOption) -> Drawable = {
+private val errorImageLoader: suspend () -> Drawable = {
     throw IllegalStateException("Must pass in an image loader to use icon() or iconPainter.")
 }
-
-private class DelegateDrawable(
-    @Volatile private var delegate: Drawable,
-    private val imageLoader: suspend (PaymentOption) -> Drawable,
-    private val paymentOption: PaymentOption,
```

Comment:
Mostly just an extraction, but slightly changed to remove the direct dependency on PaymentOption.